src/main/java/org/broadinstitute/hellbender/tools/walkers/sv/SVCleanPt1a.java

```diff
@@ -0,0 +1,439 @@
+        if (genotype.hasExtendedAttribute(GATKSVVCFConstants.RD_CN)) {
+            final int rdCN = Integer.parseInt(genotype.getExtendedAttribute(GATKSVVCFConstants.RD_CN).toString());
+            genotypeBuilder.attribute(GATKSVVCFConstants.RD_CN, rdCN + 1);
+
+            final Allele refAllele = genotype.getAllele(0);
+            final Allele altAllele = genotype.getAllele(1);
+            if (svType.equals(GATKSVVCFConstants.SYMB_ALT_STRING_DEL)) {
+                if (rdCN >= 1) genotypeBuilder.alleles(Arrays.asList(refAllele, refAllele));
+                else if (rdCN == 0) genotypeBuilder.alleles(Arrays.asList(refAllele, altAllele));
+            } else if (svType.equals(GATKSVVCFConstants.SYMB_ALT_STRING_DUP)) {
+                if (rdCN <= 1) genotypeBuilder.alleles(Arrays.asList(refAllele, refAllele));
+                else if (rdCN == 2) genotypeBuilder.alleles(Arrays.asList(refAllele, altAllele));
+                else genotypeBuilder.alleles(Arrays.asList(altAllele, altAllele));
+            }
+        }
+    }
+
+    private boolean isRevisableEvent(VariantContext variant, boolean isY) {
+        final List<Genotype> genotypes = variant.getGenotypes();
+        int[] maleCounts = new int[4];
+        int[] femaleCounts = new int[4];
+        for (final Genotype genotype : genotypes) {
+            final String sampleName = genotype.getSampleName();
+            final Integer sex = sampleSexMap.get(sampleName);
+            if (sex == null) continue;
+
+            int rdCN = (int) genotype.getExtendedAttribute(GATKSVVCFConstants.RD_CN, -1);
+            if (rdCN == -1) continue;
+
+            final int rdCNVal = Math.min(rdCN, 3);
+            if (sex == 1) {
+                maleCounts[rdCNVal]++;
+            } else if (sex == 2) {
+                femaleCounts[rdCNVal]++;
+            }
+        }
+
+        final double maleMedian = calcMedian(maleCounts);
+        final double femaleMedian = calcMedian(femaleCounts);
+        return maleMedian == 1.0 && (isY ? femaleMedian == 0.0 : femaleMedian == 2.0);
+    }
+
+    private double calcMedian(int[] counts) {
+        final int total = Arrays.stream(counts).sum();
+        if (total == 0) return Double.NaN;
+
+        final double target = total / 2.0;
+        int runningTotal = 0;
+        for (int i = 0; i < 4; i++) {
+            runningTotal += counts[i];
+            if (runningTotal == target) {
+                return i + 0.5;
+            } else if (runningTotal > target) {
+                return i;
+            }
+        }
+        throw new RuntimeException("Error calculating median");
+    }
+
+    private void processSVType(VariantContext variant, VariantContextBuilder builder) {
+        final String svType = variant.getAttributeAsString(GATKSVVCFConstants.SVTYPE, null);
+        if (svType != null && variant.getAlleles().stream().noneMatch(allele -> allele.getDisplayString().contains(GATKSVVCFConstants.ME))) {
```

Comment:
Better to parse the alleles with `GATKSVVariantContextUtils.getSymbolicAlleleSymbols()` and check for `ME`. Sometimes the alt is just `<INS:ME>`.

Reply:
Updated accordingly - thanks for the pointer.